src/components/MailboxThread.vue

```diff
@@ -367,13 +374,49 @@ export default {
 					groups.lastWeek.push(envelope)
 				} else if (date >= startOfLastMonth) {
 					groups.lastMonth.push(envelope)
+				} else if (date.getFullYear() === now.getFullYear()) {
+					const m = date.getMonth()
+					monthsMap[m] = monthsMap[m] || []
+					monthsMap[m].push(envelope)
 				} else {
-					groups.older.push(envelope)
+					const y = date.getFullYear()
+					yearsMap[y] = yearsMap[y] || []
+					yearsMap[y].push(envelope)
 				}
 			}
 
+			const orderByDate = (a, b) =>
+				this.sortOrder === 'newest' ? b.dateInt - a.dateInt : a.dateInt - b.dateInt
+
+			Object.values(groups).forEach(list => list.sort(orderByDate))
+			Object.values(monthsMap).forEach(list => list.sort(orderByDate))
+			Object.values(yearsMap).forEach(list => list.sort(orderByDate))
+
+			const groupOrder = []
+
+			const fixedGroups = ['lastHour', 'today', 'yesterday', 'lastWeek', 'lastMonth']
+			groupOrder.push(...(this.sortOrder === 'newest' ? fixedGroups : fixedGroups.reverse()))
```

Comment:
```suggestion
			groupOrder.push(...(this.sortOrder === 'newest' ? fixedGroups : fixedGroups.toReversed()))
```

`Array.reverse()` is evil as it reverse in-place, mutating the array.